internal/nix/profile.go

```diff
@@ -142,13 +181,25 @@ func (item *NixProfileListItem) AttributePath() (string, error) {
 	return attrPath, nil
 }
 
-// PackageName parses the package name from the NixProfileListItem.lockedReference
+// packageName parses the package name from the NixProfileListItem.lockedReference
 //
 // For example:
 // if NixProfileListItem.lockedReference = github:NixOS/nixpkgs/52e3e80afff4b16ccb7c52e9f0f5220552f03d04#legacyPackages.x86_64-darwin.go_1_19
 // then AttributePath = legacyPackages.x86_64-darwin.go_1_19
-// and then PackageName = go_1_19
-func (item *NixProfileListItem) PackageName() (string, error) {
+// and then packageName = go_1_19
+func (item *NixProfileListItem) packageName() (string, error) {
+	// Since unlocked references should already have absolute paths, projectDir is not needed
+	input := InputFromString(item.unlockedReference, "" /*projectDir*/)
+	if input.IsFlake() {
+		// TODO(landau): this needs to be normalized so that we can compare
+		// packages.aarch64-darwin.hello and hello and determine that they are
+		// the same package.
+		return item.unlockedReference, nil
+	}
+
+	// TODO(landau/savil): Should we use unlocked reference instead?
+	// I'm not sure we gain anything by using the locked reference. since
+	// the user specifies only the package name when installing
```

Comment:
hmm yeah, possibly. 

I can't remember why I chose to do the lockedReference, so its possible there was some scenario I ran into. But give it a try, and we can see if some edge-case doesn't work?

Reply:
would recommend a separate PR